### PR TITLE
refactor(bpp): use pluginlib to load scene module

### PR DESCRIPTION
**Pull request by satoshi-ota**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 4ff450e</samp>

This pull request refactors the behavior path planner node and its scene modules to use pluginlib for dynamic plugin loading and unloading. It also simplifies the launch and configuration of the node and its modules, and fixes some minor bugs and issues. It modifies several files, including launch files, CMakeLists.txt, header and source files, package.xml, and plugins.xml.

## Tests performed

Confirmed that bpp loaded all scene modules correctly.

```
[INFO] [component_container_mt-29]: process started with pid [145130]                                                                                                                                           
[component_container_mt-29] [INFO 1701737049.463438690] [planning.scenario_planning.lane_driving.behavior_planning.behavior_planning_container]: Load Library: /home/satoshi/pilot-auto/install/behavior_path_planner/lib/libbehavior_path_planner_lib.so
[component_container_mt-29] [INFO 1701737049.570749397] [planning.scenario_planning.lane_driving.behavior_planning.behavior_planning_container]: Found class: rclcpp_components::NodeFactoryTemplate<behavior_path_planner::BehaviorPathPlannerNode>
[component_container_mt-29] [INFO 1701737049.576008845] [planning.scenario_planning.lane_driving.behavior_planning.behavior_planning_container]: Instantiate class: rclcpp_components::NodeFactoryTemplate<behavior_path_planner::BehaviorPathPlannerNode>
[component_container_mt-29] [INFO 1701737049.722422275] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::AvoidanceModuleManager' is loaded.
[component_container_mt-29] [INFO 1701737049.728016083] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::DynamicAvoidanceModuleManager' is loaded.
[component_container_mt-29] [INFO 1701737049.731040344] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::SideShiftModuleManager' is loaded.
[component_container_mt-29] [INFO 1701737049.768085413] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::StartPlannerModuleManager' is loaded.
[component_container_mt-29] [INFO 1701737049.945528895] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::GoalPlannerModuleManager' is loaded.
[component_container_mt-29] [INFO 1701737050.164803561] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::LaneChangeRightModuleManager' is loaded.
[component_container_mt-29] [INFO 1701737050.190003251] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::LaneChangeLeftModuleManager' is loaded.
[component_container_mt-29] [INFO 1701737050.236931556] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::ExternalRequestLaneChangeRightModuleManager' is loaded.
[component_container_mt-29] [INFO 1701737050.411370194] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::ExternalRequestLaneChangeLeftModuleManager' is loaded.
[component_container_mt-29] [INFO 1701737050.600689989] [planning.scenario_planning.lane_driving.behavior_planning.behavior_path_planner]: The scene plugin 'behavior_path_planner::AvoidanceByLaneChangeModuleManager' is loaded.
```

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/0439c40c-9caf-5f14-88e9-13bebd7fcac2?project_id=prd_jt)

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/scene_module/side_shift/manager.cpp [hunk 1]
```diff
@@ -25,20 +25,22 @@
...
-  rclcpp::Node * node, const std::string & name, const ModuleConfigParameters & config)
-: SceneModuleManagerInterface(node, name, config, {})
+void SideShiftModuleManager::init(rclcpp::Node * node)
 {
+  // init manager interface
+  initInterface(node, {});
+
   SideShiftParameters p{};
 
+  std::string ns = "side_shift.";
```
Comment (kosuke55) [inline planning/behavior_path_planner/src/scene_module/side_shift/manager.cpp:35]:
```suggestion
  const std::string ns = "side_shift.";
```

Comment (kosuke55) [reply planning/behavior_path_planner/src/scene_module/side_shift/manager.cpp:35]: and others

File: planning/behavior_path_planner/src/planner_manager.cpp [hunk 2]
```diff
@@ -41,6 +42,46 @@ PlannerManager::PlannerManager(
...
+    RCLCPP_WARN_STREAM(
+      node.get_logger(),
+      "The scene plugin '" << name << "' is not found in the registered modules.");
+  } else {
+    manager_ptrs_.erase(it, manager_ptrs_.end());
+    processing_time_.erase(name);
+    RCLCPP_INFO_STREAM(node.get_logger(), "The scene plugin '" << name << "' is unloaded.");
+  }
+}
+
```
Comment (kosuke55) [inline planning/behavior_path_planner/src/planner_manager.cpp:84]: just a question. 
this is not currently used, but is this a mechanism to dynamically disable modules in the future?

Comment (satoshi-ota) [reply planning/behavior_path_planner/src/planner_manager.cpp:84]: YES.

File: planning/behavior_path_planner/src/scene_module/side_shift/manager.cpp (hunk 1 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/scene_module/side_shift/manager.cpp]: @kosuke55 Fixed in https://github.com/autowarefoundation/autoware.universe/pull/5771/commits/81332c9c3449ec0069c81597bd2435b5a439f83b.

**Review by kosuke55 — APPROVED** [review]
LGTM! thanks!"

**Pull request merged**